### PR TITLE
log4cxx 0.12.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by cho-m** [comment]
Need to switch to CMake: https://logging.apache.org/log4cxx/latest_stable/build-cmake.html

Autotool no longer supported: https://github.com/apache/logging-log4cxx/commit/bce24884809bd157cf2f85ef243d658e406a4259

File: Formula/log4cxx.rb [hunk 1]
```diff
@@ -14,24 +14,21 @@ class Log4cxx < Formula
...
-                          # Docs won't install on macOS
-                          "--disable-doxygen",
-                          "--with-apr=#{Formula["apr"].opt_bin}",
-                          "--with-apr-util=#{Formula["apr-util"].opt_bin}"
-    system "make", "install"
+    mkdir "build" do
+      system "cmake", "..",
+                      *std_cmake_args,
+                      "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                      "-DCMAKE_CXX_STANDARD=17",
```
Comment (carlocab) [inline Formula/log4cxx.rb:27]: Why is this needed?

Comment (chenrui333) [reply Formula/log4cxx.rb:27]: It is recommended to build with cxx17
https://github.com/apache/logging-log4cxx/blob/master/CMakeLists.txt#L38-L42

Comment (carlocab) [reply Formula/log4cxx.rb:27]: Let's leave it out, and let the upstream `CMakeLists.txt` handle it for us. They'll set it to the correct value as long as we don't set it.

File: Formula/log4cxx.rb [hunk 2]
```diff
@@ -14,24 +14,21 @@ class Log4cxx < Formula
...
-                          "--prefix=#{prefix}",
-                          # Docs won't install on macOS
-                          "--disable-doxygen",
-                          "--with-apr=#{Formula["apr"].opt_bin}",
-                          "--with-apr-util=#{Formula["apr-util"].opt_bin}"
-    system "make", "install"
+    mkdir "build" do
+      system "cmake", "..",
+                      *std_cmake_args,
+                      "-DCMAKE_INSTALL_RPATH=#{rpath}",
```
Comment (carlocab) [inline Formula/log4cxx.rb:26]:
```suggestion
```
We don't need to set this either, as it doesn't look like they make use of `RPATH` anywhere.

**Review by carlocab — APPROVED** [review]
